source/02_algo/model_based_rl_eng.rst

```diff
@@ -0,0 +1,147 @@
+  :align: center
+  :scale: 55%
+
+The environment model can generally be abstracted mathematically into a state transition function and a reward function.
+After learning the environment model, ideally, the agent does not need to interact with the real environment, but only in the simulated environment, which maximizes the cumulative discount reward through the RL algorithm to obtain the optimal policy.
+
+
+Problem Definition and Research Motivation
+----------------------------------------------
+
+In general, the problems of Model-Based RL research can be divided into two categories: one is how to learn an accurate environment model, and the other is how to use the environment model for policy optimization.
+
+**How to build an accurate environment model?** 
+
+Model learning mainly emphasizes the process of building an environment model by the Model-Based RL algorithm. For example, 
+
+  - `World Model <https://worldmodels.github.io/>`_ [3]_ proposes an environment model based on unsupervised learning and uses this model to transfer tasks from simulation to reality.
+  - `I2A <https://arxiv.org/abs/1707.06203>`_ [4]_ proposes an imagination-augmented-based model structure, based on which the future trajectory is predicted, and the trajectory information is encoded to assist policy learning.
+
+  But Model-Based RL also has several problems in the model learning part, for example,
+
+  - There will be errors in the environment model, and with the iterative interaction between the algorithm and the environment model, the accumulated error will become larger and larger, making it difficult for the algorithm to converge to the optimal solution.
+  - The environment model lacks generality, and every time a problem is changed, the model must be re-modeled.
+
+**How to use the environment model for policy optimization?**
+
+Model utilization mainly emphasizes that Model-Based RL algorithms use environmental models to assist policy learning, such as model-based planning or model-based policy learning.
+
+  - Both `ExIt <https://arxiv.org/abs/1705.08439>`_ [5]_ and `AlphaZero <https://arxiv.org/abs/1712.01815>`_ [6]_ are based on expert iteration and Monte Carlo tree search methods to learn strategies.
+  - `POPLIN <https://openreview.net/forum?id=H1exf64KwH>`_ [7]_ does online planning based on the environment model, and proposes optimization ideas for action space and parameter space respectively.
+  - `M2AC <https://arxiv.org/abs/2010.04893>`_ [8]_ proposes a mask mechanism based on model uncertainty, which enhances policy improvement.
+
+
+Research Direction
+--------------------
+
+The papers of Model-Based RL in recent years have been sorted out and summarized in `awesome-model-based-RL <https://github.com/opendilab/awesome-model-based-RL>`_ [1]_.
+One of the most classic Model-Based RL algorithms is Dyna-type reinforcement learning, which is a type of architecture that combines Model-Based RL and Model-Free RL.
```

Comment:
One of the most classic Model-Based RL algorithms is **_Dyna-style_** reinforcement learning, which is a type of algorithm that combines Model-Based RL and Model-Free RL.